app/components-react/shared/ModalLayout.tsx

```diff
@@ -0,0 +1,77 @@
+import React, { ReactNode } from 'react';
+import { useVuex } from '../hooks';
+import { Services } from '../service-provider';
+import { getOS, OS } from '../../util/operating-systems';
+import classNames from 'classnames';
+import { $t } from '../../services/i18n';
+import css from './ModalLayout.m.less';
+
+export interface IModalLayoutProps {
+  hasTitleBar?: boolean;
+  showControls?: boolean;
+  showDone?: boolean;
+  disableDone?: boolean;
+  onSubmit?: (...args: unknown[]) => unknown;
+}
+type TProps = IModalLayoutProps & { children: ReactNode };
+
+/**
+ * A modal layout for showing dialogs
+ */
+export function ModalLayout(partialProps: TProps) {
+  // inject services
+  const { WindowsService, CustomizationService } = Services;
+
+  // define default props
+  const props = {
+    hasTitleBar: true,
+    showCancel: true,
+    showControls: true,
+    showDone: true,
+    onSubmit: () => {},
+    ...partialProps,
+  };
+
+  // define a reactive state
+  const state = useVuex(() => ({ currentTheme: CustomizationService.currentTheme }));
```

Comment:
also is this referencing a `view` or direct service state?

Reply:
We can use `state`, `globalState` or `localState`. But since this component doesn't have a local state I used a simple `state` word.
The pattern I used is from the Redux library. They call the returning object as `selectedData`